src/terminal-util.c

```diff
@@ -707,6 +708,56 @@ terminal_util_add_proxy_env (GHashTable *env_table)
 	g_object_unref (settings_socks);
 }
 
+/**
+ * terminal_util_hyperlink_uri_label:
+ * @uri: a URI
+ *
+ * Formats @uri to be displayed in a tooltip.
+ * Performs URI-decoding and converts IDN hostname to UTF-8.
+ *
+ * Returns: (transfer full): The human readable URI as plain text
+ */
+char *terminal_util_hyperlink_uri_label (const char *uri)
+{
+	char *unesc = NULL;
+	gboolean replace_hostname;
+
+	if (uri == NULL)
+		return NULL;
+
+	unesc = g_uri_unescape_string(uri, NULL);
+	if (unesc == NULL)
+		unesc = g_strdup(uri);
+
+	if (g_ascii_strncasecmp(unesc, "ftp://", 6) == 0 ||
+		g_ascii_strncasecmp(unesc, "http://", 7) == 0 ||
+		g_ascii_strncasecmp(unesc, "https://", 8) == 0 ||
+		g_ascii_strncasecmp(unesc, "mailto://", 9) == 0)
```

Comment:
mailto doesn't use slashes, should be `"mailto:", 7`, but I don't know if mail addresses are ever IDN. I guess they can be anyway. I can confirm against the RFC next time I have a chance.

Reply:
Year, my fault. I took a looked at the code more attentive and it won't be work correctly with email addresses anyway. I'll fix it in a separate patch